plugins/ltac2/tac2def.ml

```diff
@@ -0,0 +1,86 @@
+(************************************************************************)
+(*         *   The Coq Proof Assistant / The Coq Development Team       *)
+(*  v      *         Copyright INRIA, CNRS and contributors             *)
+(* <O___,, * (see version control and CREDITS file for authors & dates) *)
+(*   \VV/  **************************************************************)
+(*    //   *    This file is distributed under the terms of the         *)
+(*         *     GNU Lesser General Public License Version 2.1          *)
+(*         *     (see LICENSE file for the text of the license)         *)
+(************************************************************************)
+
+open Proofview
+open Tac2ffi
+
+(* Local shortcuts. *)
+type v = Tac2ffi.valexpr
+type 'r of_v = v -> 'r
+type 'r to_v = 'r -> v
+type 'a with_env = Environ.env -> Evd.evar_map -> 'a
+
+let value : valexpr repr = make_repr Fun.id Fun.id
+
+type (_, _, _) spec =
+  | T : 'r to_v -> (v tactic, 'r tactic, 'r) spec
+  | V : 'r to_v -> (v tactic, 'r, 'r) spec
+  | E : 'r to_v -> (v tactic, 'r with_env, 'r) spec
+  | G : 'r to_v -> (v tactic, Goal.t -> 'r, 'r) spec
+  | F : 'a of_v * ('t, 'f, 'r) spec -> (v -> 't , 'a -> 'f, 'r) spec
```

Comment:
These names are not very descriptive.

Reply:
True, but does it matter since they are not part of the interface?
Given the types, it is kind of obvious what they do, and using the wrong constructor would not type-check.